### ISSUE TITLE
Allow Bot Merging Permission to Protected Branches

### ISSUE THREAD

**Issue by rogerluan** [bobvanderlinden/probot-auto-merge#805]
Hello!

I was just configuring `probot-auto-merge` in a repo of mine and I was facing this error after all merging conditions were met:

![image](https://user-images.githubusercontent.com/8419048/89606117-99a08b80-d845-11ea-86aa-bc72beb1cb80.png)

The PR was green, the checks had passed, the PR was approved, etc.

Then I realized that since `master` is a protected branch, I had to enable `probot-auto-merge` bot to merge to master.

Once I did this, it worked. However, the error message seen before was very misleading and led me to waste time investigating something unrelated. Would it be possible to improve this error message or, better yet, add this extra step to the documentation? Or should this not be a requirement at all? There's a related issue here https://github.com/bobvanderlinden/probot-auto-merge/issues/85 but it's unclear what's the desired behavior (also, that issue is almost 2 years old).

Appreciate any input! Thank you for this awesome bot!

**Comment by bobvanderlinden** [comment]
Oh, that's very good to know! `MergeStateStatus.BLOCKED` is very scarcely documented. See https://docs.github.com/en/graphql/reference/enums#mergestatestatus

> **BLOCKED**
> 
> The merge is blocked.

Thanks for figuring this one out! I'll make a PR that updates explains a bit more why the pull request merge is blocked.

**Comment by bobvanderlinden** [comment]
Can you confirm that the following was what resolved the issue for you:

![Branch protection rule 2020-08-07 10-07-24](https://user-images.githubusercontent.com/6375609/89624021-ef048900-d895-11ea-9381-c4f4cfb96abf.png)

**Comment by rogerluan** [comment]
Yup, doing exactly that fixed my issue @bobvanderlinden !

**Comment by rogerluan** [comment]
@bobvanderlinden circling back on this issue, as I found out more information about it.

Turns out that the blocked status https://docs.github.com/en/graphql/reference/enums#mergestatestatus is returned whenever **any** branch protection rule isn't met. I initially thought the only issue would be the `"restrict who can push to matching branches"` option, however, I just realized that my rules included "2+ reviewers" and I only had 1 reviewer and the bot refused to merge. In other words, it will return "BLOCKED" status whenever the Pull Request mergeable state is not "green" (in GitHub website UI).

I'm just adding more info here for future readers and have some sort of documentation of my finds :) I think our current message is appropriate as is and no further changes are needed.
